pygmt/helpers/decorators.py

```diff
@@ -811,3 +811,61 @@ def new_module(*args, **kwargs):
         return new_module
 
     return deprecator
+
+
+def check_data_input_order(deprecate_version, remove_version):
+    """
+    Decorator to raise a FutureWarning if the order of data input parameters
+    changes and positional arguments are passed.
+
+    Parameters
+    ----------
+    deprecate_version : str
+        The PyGMT version when the order of data input parameters is changed.
+    remove_version : str
+        The PyGMT version when the deprecation warning should be removed.
+
+    Examples
+    --------
+    >>> @check_data_input_order("v0.0.0", "v9.9.9")
+    ... def module(data=None, x=None, y=None, z=None, **kwargs):
+    ...     "A module that prints the arguments it received"
+    ...     print(f"data={data}, x={x}, y={y}, z={z}")
+    >>> module(data="table.txt")
+    data=table.txt, x=None, y=None, z=None
+    >>> module(x=0, y=1, z=2)
+    data=None, x=0, y=1, z=2
+    >>> with warnings.catch_warnings(record=True) as w:
+    ...     module(0, 1, 2)
+    ...     assert len(w) == 1
+    ...     assert issubclass(w[0].category, FutureWarning)
+    ...
+    data=0, x=1, y=2, z=None
+    """
+
+    def data_input_order_checker(module_func):
+        """
+        The decorator that creates the new function to check if positional
+        arguments are passed.
+        """
+
+        @functools.wraps(module_func)
+        def new_module(*args, **kwargs):
+            """
+            New module instance that raises a warning if positional arguments
+            are passed.
+            """
+            if len(args) != 0:  # positional arguments are used
```

Comment:
Don't think args can be negative, but this looks better I think :slightly_smiling_face:
```suggestion
            if len(args) > 0:  # positional arguments are used
```